### PR TITLE
[build] Make travis logs not as long

### PR DESCRIPTION
**Pull request by ericl**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ matrix:
 
       # Test Bazel build
       - rm -rf build
-      - ./ci/travis/install-bazel.sh
+      - out=`mktemp`; ./ci/travis/install-bazel.sh >$out 2>&1 || cat $out
```
Comment (raulchen) [inline .travis.yml]: What is the purpose of this change? It doesn't make the log shorter, right?
Also, it'd a bit nicer to use a function. As this pattern is repeated many times.

File: .travis.yml [hunk 2]
```diff
@@ -157,20 +157,20 @@ script:
...
+  - python/ray/rllib/tests/run_silent.sh test_filters.py
+  - python/ray/rllib/tests/run_silent.sh test_optimizers.py
+  - python/ray/rllib/tests/run_silent.sh test_evaluators.py
 
   # ray tests
   # Python3.5+ only. Otherwise we will get `SyntaxError` regardless of how we set the tester.
-  - python -c 'import sys;exit(sys.version_info>=(3,5))' || python -m pytest -v --durations=10 python/ray/experimental/test/async_test.py
-  - python -m pytest -v --durations=30 python/ray/tests
+  - python -c 'import sys;exit(sys.version_info>=(3,5))' || python -m pytest --durations=5 python/ray/experimental/test/async_test.py
+  - python -m pytest --durations=10 python/ray/tests
```
Comment (raulchen) [inline .travis.yml]: We may want to keep the `-v`. With it, I can easily find failed test by searching uppercased `FAILED`. Plus, it won't make the log too long.

File: .travis.yml (hunk 1 above)
Comment (ericl) [reply .travis.yml]: @raulchen here all output is suppressed unless the script exits with failure.

Agree about a function. I can make a script like run_silent.sh for other tests.

File: .travis.yml (hunk 2 above)
Comment (ericl) [reply .travis.yml]: I think this should still be the case. The goal is to suppress output for successful tests. Let me know if you have better ideas on how to do this.

File: .travis.yml (hunk 1 above)
Comment (raulchen) [reply .travis.yml]: > @raulchen here all output is suppressed unless the script exits with failure.

I see. This would be useful. However, there's a drawback of doing this. When a command hangs, there will be no output at all. And it's difficult to find out where the command is stuck at. Maybe in the `run_silent.sh` script, we can print the file if there's no output to the file for 5 minutes?

Also, another thing to notice is that travis will kill a job if there's no output for 10 minutes. We need to make sure this won't happen if a command simply needs more than 10 minutes.

Comment (ericl) [reply .travis.yml]: It turns out you can use `timeout` to enforce a time limit. I set it to 30 minutes here.

To avoid the travis timeout, we could also set it to 9 minutes, if we are sure each build step takes less than that time.

File: .travis.yml (hunk 2 above)
Comment (raulchen) [reply .travis.yml]: AFAIK, the `-v` option only prints the progress with test names. E.g.,
```
test/actor_test.py::test_actor_init_error_propagated PASSED                      [  1%]
test/actor_test.py::test_keyword_args PASSED                                     [  2%]
test/actor_test.py::test_variable_number_of_args PASSED                          [  4%]
test/actor_test.py::test_no_args PASSED                                          [  5%]
test/actor_test.py::test_no_constructor PASSED                                   [  7%]
```

Without `-v`, the progress will be just a series of dots. E.g.,
```
test/actor_test.py .....
```

This doesn't seems to have anything to do with the output.

File: .travis.yml (hunk 1 above)
Comment (raulchen) [reply .travis.yml]: Yeah, just saw you already did this.

Maybe we can print `Command xxx is still running.` every 9 minutes.

File: .travis.yml (hunk 2 above)
Comment (ericl) [reply .travis.yml]: Right, that's the goal. If we print PASSED for each test, that adds thousands of lines of output, which is quite slow to render.

The dots are significantly more compact, and actually makes it easier to search for "failed" tests (actually there is no need to search since the only output is failed test logs).

**Comment by ericl** [comment]
Since "timeout" is not available on OSX, rewrote it to use just a normal child process.

**Review by pcmoritz — APPROVED** [review]
LGTM

**Comment by ericl** [comment]
Tests look good. Let's merge an iterate on this, I think its already a good improvement

File: .travis.yml (hunk 2 above)
Comment (robertnishihara) [reply .travis.yml]: This is going to be an issue for hanging tests, right? When a test hangs and Travis times out, we won't know which test was causing the problem.

Comment (raulchen) [reply .travis.yml]: I believe rendering being slow is due to the build scripts.  Using `-v` should only add a few hundred lines. 
And I think what Robert said is also an issue.
